.dotnet/src/Utility/MultipartFormDataBinaryContent.cs

```diff
@@ -0,0 +1,155 @@
+    }
+
+    public string ContentType
+    {
+        get
+        {
+            Debug.Assert(_multipartContent.Headers.ContentType is not null);
+
+            return _multipartContent.Headers.ContentType!.ToString();
+        }
+    }
+
+    internal HttpContent HttpContent => _multipartContent;
+
+    public void Add(Stream stream, string name, string fileName = default)
+    {
+        Add(new StreamContent(stream), name, fileName);
+    }
+
+    public void Add(string content, string name, string fileName = default)
+    {
+        Add(new StringContent(content), name, fileName);
+    }
+
+    public void Add(int content, string name, string fileName = default)
+    {
+        // https://learn.microsoft.com/en-us/dotnet/standard/base-types/standard-numeric-format-strings#GFormatString
+        string value = content.ToString("G", CultureInfo.InvariantCulture);
+        Add(new StringContent(value), name, fileName);
+    }
+
+    public void Add(double content, string name, string fileName = default)
+    {
+        // https://learn.microsoft.com/en-us/dotnet/standard/base-types/standard-numeric-format-strings#GFormatString
+        string value = content.ToString("G", CultureInfo.InvariantCulture);
+        Add(new StringContent(value), name, fileName);
+    }
+
+    public void Add(byte[] content, string name, string fileName = default)
+    {
+        Add(new ByteArrayContent(content), name, fileName);
+    }
+
+    public void Add(BinaryData content, string name, string fileName = default)
+    {
+        Add(new ByteArrayContent(content.ToArray()), name, fileName);
+    }
+
+    private void Add(HttpContent content, string name, string fileName)
+    {
+        if (fileName is not null)
+        {
+            AddFileNameHeader(content, name, fileName);
+        }
+
+        _multipartContent.Add(content, name);
+    }
+
+    private static void AddFileNameHeader(HttpContent content, string name, string filename)
+    {
+        // Add the content header manually because the default implementation
+        // adds a `filename*` parameter to the header, which RFC 7578 says not
+        // to do.  We are following up with the BCL team per correctness.
+        ContentDispositionHeaderValue header = new("form-data")
+        {
+            Name = name,
+            FileName = filename
+        };
+        content.Headers.ContentDisposition = header;
+    }
+
+    private static string CreateBoundary()
+    {
+        Span<char> chars = new char[70];
+
+        byte[] random = new byte[70];
+        _random.NextBytes(random);
+
+        // The following will sample evenly from the possible values.
+        // This is important to ensuring that the odds of creating a boundary
+        // that occurs in any content part are astronomically small.
+        int mask = 255 >> 2;
+
+        Debug.Assert(_boundaryValues.Length - 1 == mask);
+
+        for (int i = 0; i < 70; i++)
```

Comment:
@KrzysztofCwalina, I would love your thoughts on whether there's a more performance-efficient way to accomplish this.

Reply:
I am glad you asked. @tg-msft and @Petermarcu did a lot of thinking here. If I remember correctly, the following was the most performant way:
```csharp
static void GenerateBoundary(Span<byte> buffer)
        {
            if (buffer.Length != 64 && buffer.Length != 70) { throw new ArgumentException(); }
            ReadOnlySpan<char> format = ['N'];
            Guid.NewGuid().TryFormat(buffer[..32], out var _, format);
            Guid.NewGuid().TryFormat(buffer[32..64], out var _, format);
            if (buffer.Length == 70)
            {
                // It won't write if the slice is too small,
                // so we just write over the second GUID
                Guid.NewGuid().TryFormat(buffer[38..70], out var _, format);
            }
        }
```

but maybe we can simplify to 
```csharp
var twoGuids = Guid.NewGuid().ToString("N") + Guid.NewGuid().ToString("N");
        byte[] boundary = Encoding.ASCII.GetBytes(twoGuids);
        return boundary;
```
It's not as unique, but it's simple and probably unique enough.

Reply:
@KrzysztofCwalina, I went digging for that quote on the size of the universe (it is somewhere in [here](https://seattle.bibliocommons.com/v2/record/S30C3644590)), but couldn't put my hands on the exact one I was thinking of.  From this snip, it looks like it may have been a technical term meaning the size of the set, and not a reference to the number of quarks in the universe or some other astronomically large number.  Ah well.  🙃

![image](https://github.com/joseharriaga/openai-in-typespec/assets/21269347/5b120d52-02bf-447a-80da-238bc4e3aec3)